Artesian/Artesian.SDK/Service/Query/IQueryWithFill.cs

```diff
@@ -0,0 +1,16 @@
+﻿// Copyright (c) ARK LTD. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for
+// license information. 
+using NodaTime;
+
+namespace Artesian.SDK.Service
+{
+    interface IQueryWithFill<T>: IBaseQuery<T>
```

Comment:
Name is misleading as it also include RelativePeriod.

Either rename it, or split in 2 interfaces or any other clean solution :)

Reply:
From reading the wiki it is indicated that Auction curves are not timeseries . This difference then would still not indicate a more appropriate rename to the interface given that the implementation differs in extraction range options , transform and filler options. Also given that in the future other data kinds may have alternative implementations that contradict this. There does not seem to be a clear and accurate rename available.

Splitting into 2 interfaces is this reverting to the original IQuery implementation and creating a separate "IQueryAuction" for Auctions that duplicate the existing and necessary functionality? Is there a case that if in future another data kind would potentially require this duplication again? This would probably still be the cleanest solution.

Reply:
Splitting in 2 means having IQueryFill with only Fill related methods and
IQueryRelative for Relative methods. The problem is IQueryWithFill that
also have methods related to Relative extraction.